### ISSUE TITLE
Textarea improvements and additions

### ISSUE THREAD

**Issue by crisbeto** [angular/material#7649]
TL;DR from the [Google Doc](https://docs.google.com/document/d/1NDtBURsRIkswgrjdDEcB1wzt-asBLnezXiZ0M_ryYPs/edit?usp=sharing):

We should look into some improvements to the `textarea` directive:

* We should support growing and shrinking according to the content. The current implementation doesn't shrink down if the `rows` are specified.
* We should support a minimum amount of rows. The current solution doesn't respect the `rows` unless there's enough content to force the size beyond the amount of rows.
* We should have the ability to resize the textarea.

CC @EladBezalel, @DevVersion, @ThomasBurleson

**Comment by DevVersion** [comment]
@crisbeto @EladBezalel - Someone mentioned that we should support an asterisk on the textarea as well. 

I can see the point and see no restrictions in the specs.

**Comment by crisbeto** [comment]
Sound good to me.

**Comment by EladBezalel** [comment]
@DevVersion please add to docs
